sdk-setup/src/sdk-make-qmltypes

```diff
@@ -0,0 +1,1554 @@
+            "Name:" tag is selected.
+
+        Normalized Git URL
+            This must be in the form
+                SERVICE ':' PATH
+            E.g. "git.merproject.org:mer-core/ssu". The PATH must be relative.
+
+        Git Branch
+            The Git branch to check out
+
+        Binary Packages
+            A comma separated list of the binary packages which provide the QML
+            modules to dump type information for. Each package can be listed
+            ether with (a) full name, (b) name starting with "-" which will be
+            treated as a suffix to the Package Name or (c) just "-" which
+            stands for name equal to the Package Name.
+
+    Example configuration file
+
+        # PACKAGE  REPOSITORY                       BRANCH  BINARIES
+        ssu        git.merproject.org:mer-core/ssu  master  -declarative
+        mpris-qt5  github.com:nemomobile/qtmpris    master  -qml-plugin
+
+    A remote configuration file may be specified with an HTTP(S)/FTP(S) URL in
+    which case curl(1) will be used for download. Note that curl does not
+    enable netrc(5) based authentication by default - add '--netrc' to your
+    ~/.curlrc if you wish to use netrc authentication.
+
+    Alternatively the configuration file can use the CSV format. See '--csv'.
+    This is useful e.g. for keeping the configuration in an Ethercalc
+    spreadsheet:
+
+        sdk-make-qmltypes --batch --csv
+            https://ethercalc.example.com/_/<SPREADSHEET-ID>/csv \
+            ~/projects
+
+
+OVERVIEW OF BATCH OPERATION
+    Each package is processed gradually, with these possible states:
+
+    TODO
+        Processing not started yet
+
+    CLONED
+        Reference repository was located under PROJECTS_DIR and it was cloned
+        under the working directory
+
+    MAKEFILES-READY
+        Depending on the project either qmake or CMake was run to produce
+        Makefiles
+
+    INSTALLED
+        Binary packages were installed to the emulator
+
+    DONE
+        The qmltypes files were updated by running \`make qmltypes\` in every
+        directory that contains at least one '*.qmltypes' file tracked by Git
+
+    FAIL
+        Package was marked as failed and will be excluded from processing
+
+LIMITATIONS
+    Only qmake- and CMake-based projects are supported.
+
+    Batch operation is only supported under Docker-based build engine running
+    on a Linux host.
+END
+}
+
+debug()
+{
+    :
+}
+
+info()
+{
+    echo "$*" >&2
+}
+
+warning()
+{
+    echo "Warning: $*" >&2
+}
+
+fatal()
+{
+    echo "Fatal: $*" >&2
+}
+
+configure()
+{
+    if ! inside_build_engine; then
+        fatal "This tool can be used inside SDK build engine only."
+        return 1
+    fi
+
+    if ! behind_sfdk; then
+        fatal "This tool can be only used from session opened with sfdk."
+        return 1
+    fi
+
+    if [[ $OPT_BATCH ]]; then
+        if ! python3 -c 'import giturlparse' &>/dev/null; then
```

Comment:
How big task would it be to package giturlparse, and make it a requirement?

Reply:
Small I believe. I just didn't pay much attention to this as it is only needed for the batch mode, which is not available (advertised) to regular users through sfdk.  I would package it later when utilizing the batch mode execution under sdk-test-suite.

Reply:
I'll add a task in bugzilla about that.

Reply:
Thank you